undertow/src/main/java/org/wildfly/extension/undertow/security/AuditNotificationReceiver.java

```diff
@@ -101,7 +101,13 @@ private static String deriveUsefulInfo(HttpServletRequest httpRequest) {
         Enumeration<?> enparam = httpRequest.getParameterNames();
         while (enparam.hasMoreElements()) {
             String paramName = (String) enparam.nextElement();
-            String[] paramValues = httpRequest.getParameterValues(paramName);
+            String[] paramValues;
+            if (paramName.equals("j_password")) {
+                paramValues = new String[1];
+                paramValues[0] = "***";
```

Comment:
This can be reduced down to a single line:

paramValues = new String[] {"***"};